### PR TITLE
Add unit test overview

### PR DESCRIPTION
**Pull request by milesgranger**
- [x] Part of https://github.com/dask/dask/issues/10732
- [x] Tests added / passed
- [x] Passes `pre-commit run --all-files`

Mostly taken from dask/distributed with small adjustments

### CODE REVIEW

**Comment by GPUtester** [comment]
Can one of the admins verify this patch?

Admins can comment `ok to test` to allow this one PR to run or `add to allowlist` to allow all future PRs from the same author to run.

**Comment by phofl** [comment]
add to allowlist

**Comment by phofl** [comment]
lets see if I have sufficient permissions for this

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -108,8 +108,39 @@ jobs:
...
+        shell: bash
+
+      - name: Upload test results
+        # ensure this runs even if pytest fails
+        if: >
+          always() &&
+          (steps.run_tests.outcome == 'success' || steps.run_tests.outcome == 'failure')
+        uses: actions/upload-artifact@v4
+        with:
+          name: ${{ env.TEST_ID }}
```
Comment (crusaderky) [inline .github/workflows/tests.yml:132]: Name collision with the 'upload environment file' action

**Review by crusaderky — CHANGES_REQUESTED** [review]
It doesn't seem to work:

https://github.com/crusaderky/dask/pull/25
https://github.com/crusaderky/dask/pull/25/checks?check_run_id=20354061965
https://github.com/crusaderky/dask/actions/runs/7478648117/job/20354047349

> 2024-01-10 17:31:53 +0000 - publish - WARNING - Could not find any JUnit XML files for artifacts/**/*.xml

I think the issue is caused by the name collision below

File: .github/workflows/tests.yml (hunk 1 above)
Comment (crusaderky) [inline .github/workflows/tests.yml:132]:
```suggestion
          name: test-results-${{ env.TEST_ID }}
```

**Comment by milesgranger** [comment]
Ah drats, my apologies @crusaderky I should have tested this myself. Working now: 
- https://github.com/milesgranger/dask/pull/1
- https://github.com/milesgranger/dask/actions/runs/7485268781

**Review by crusaderky — APPROVED** [review]
Looks good. 

ATM it's quite unusable though, because the report is flooded with failures from the dask-expr environment. Not sure if we want to spend time to omit that env for now or if we just want to wait for it to become green?

cc @fjetter @phofl

**Pull request merged**